$MirOS: ports/mailnews/pine/patches/patch-pico_osdep_spell_unx,v 1.1.7.1 2005/03/18 15:49:48 tg Exp $
$OpenBSD: patch-pico_osdep_spell_unx,v 1.2 2003/05/27 23:38:33 brad Exp $
--- pico/osdep/spell.unx.orig	Thu Dec 12 22:18:30 2002
+++ pico/osdep/spell.unx	Wed May  4 23:28:08 2005
@@ -84,6 +84,14 @@ spell(f, n)
     setimark(0, 1);
     emlwrite("Checking spelling...", NULL); 	/* greetings */
 
+    if((sp = (char *)getenv("SPELL")) == NULL)
+      sp = SPELLER;
+
+    /* This code only works with "spell", which doesn't exist, and
+       alt_speller is only set via command line -s argument. */
+    if (strstr(sp, "ispell") != NULL)
+      alt_speller = sp;
+
     if(alt_speller)
       return(alt_editor(1, 0));			/* f == 1 means fork speller */
 
@@ -91,9 +99,6 @@ spell(f, n)
 	emlwrite("Can't write temp file for spell checker", NULL);
 	return(-1);
     }
-
-    if((sp = (char *)getenv("SPELL")) == NULL)
-      sp = SPELLER;
 
     if(fexist(sp, "x", (off_t *)NULL) != FIOSUC){
         emlwrite("\007Spell-checking file \"%s\" not found", sp);
